addons/point_of_sale/models/stock_picking_type.py

```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+# Copyright (C) 2004-2008 PC Solutions (<http://pcsol.be>). All Rights Reserved
+from odoo import fields, models, api, _
+from odoo.exceptions import UserError
+
+
+class StockPickingType(models.Model):
+    _inherit = 'stock.picking.type'
+
+    @api.multi
+    def unlink(self):
+        for picking_type in self:
+            pos_session_ids = self.env['pos.session'].search([('config_id.picking_type_id', '=', picking_type.id), ('state', '!=', 'closed')]).ids
```

Comment:
same as above, doing a search with `foo_id.bar_id = ...` is a bit costly, especially if there is many records (like on a pos.session), better do it in the other way (picking -> config -> sessions)